lib/ide/parser/src/api.rs

```diff
@@ -15,6 +16,9 @@ pub use ast::Ast;
 pub trait IsParser : Debug {
     /// Parse program.
     fn parse(&mut self, program:String, ids:IdMap) -> Result<Ast>;
+
+    /// Parse a module content that contains idmap and metadata.
+    fn parse_as_module(&mut self, program:String) -> Result<ModuleWithMetadata>;
```

Comment:
`parse_with_metadata`
This is not module in any way more than the other call.
(actually they all return module)